jetty-util/src/main/java/org/eclipse/jetty/util/thread/QueuedThreadPool.java

```diff
@@ -234,30 +234,29 @@ protected void doStop() throws Exception
         // If stop timeout try to gracefully stop
         long timeout = getStopTimeout();
         BlockingQueue<Runnable> jobs = getQueue();
-        if (timeout > 0)
-        {
-            // Fill the job queue with noop jobs to wakeup idle threads.
-            for (int i = 0; i < threads; ++i)
-            {
-                jobs.offer(NOOP);
-            }
 
-            // try to let jobs complete naturally for half our stop time
-            joinThreads(System.nanoTime() + TimeUnit.MILLISECONDS.toNanos(timeout) / 2);
+        // Fill the job queue with noop jobs to wakeup idle threads.
+        for (int i = 0; i < threads; ++i)
+            jobs.offer(NOOP);
```

Comment:
Since we are reviewing this, I don't think this loop guarantees that all threads are woken up.
On a single core, it may be that a single thread wakes up and consumes all the NOOP jobs in the queue.

Reply:
I think you are right.  The Runner used to have a `while(isRunning)` loop, so once woken it would check that state.  Now it is only checking the equivalent state only if it doesn't have a job.   So maybe we should always check the running state again?

Reply:
Ah no, the running state is already always tested - either in addCounts or explicitly if there is no jobs.  It might be clearer to always call addCounts.... let me see...

Reply:
@sbordet I've pushed a refactored version that I think makes this clearer... and I am 95% sure is exactly the same logic.
@lachlan-roberts can you have a look and triple check the latest checking doesn't actually change the idle thread count logic.